src/components/DataTable2/DataTable2.jsx

```diff
@@ -418,6 +399,33 @@ function DataTableCardPill2({isLoading, rowHeight}) {
 	)
 }
 
+// Renders a filter badge with custom or default content
+function BadgeRenderer({ item, value, isLoading, onRemove }) {
+	const { getFilterField, getCustomPill } = useDataTableContext();
+	const { t } = useTranslation();
+
+	// Get custom pill
+	const CustomBadge = getCustomPill(item.substring(1));
+
+	return (
+		<Badge color='primary' pill>
+			{CustomBadge ? CustomBadge : `${getFilterField(item.substring(1))}: ${value}`}
```

Comment:
Hmmm, when I think of it, this is something like half solution - if you are adding a custom badge, you should be doing that as a whole thing (also with Remove etc. and have the logic in your parent code

```
return(
     CustomBadge ? CustomBadge : <Badge color='primary' pill> .... </Badge>
)
```